clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -1897,6 +1897,35 @@ public void testConsumerPositionUpdatedWhenSkippingAbortedTransactions() {
         assertEquals(currentOffset, (long) subscriptions.position(tp0));
     }
 
+    @Test
+    public void testFetchWithAuthenticationFailure() {
```

Comment:
I was looking for somewhere below where we are catching the `AuthenticationException`, but couldn't find one. Should there be one?

Reply:
Yes, the reason is with the current `Fetcher.sendFetches()` implementation the listener swallows the exceptions. I couldn't find a way to detect the authentication exception that is occurred. As an alternative I'll replace that API call with a different API that throws the exception. Please let me know if that's satisfactory, or if I missed something.